content/1.guides/2.your-first-app.md

```diff
@@ -0,0 +1,244 @@
+---
+title: Your first Genie app
+description: Short guide on creating a simple app with a form triggering some computations.
+---
+
+# Your first Genie app
+
+The general assumption in Genie apps is that you have some code performing scientific computations and want to share it online. With Genie Framework, you can encapsulate your code into a web app, making it readily available and interactive for users via a web browser.
+
+This guide will show you how to create a simple app with a page in which the user can enter a value and trigger a computation, with the result being displayed afterwards. In particular, the app will take a value `N`, generate `N` random numbers, and return their mean as in this preview:
+
+<img class="border-gray-300 border-2" style="display:block;width:90%;max-width:40%;margin-left:auto;margin-right:auto" src="/assets/guides/first-genie-app/preview.gif">
+
+
+### Running the app
+
+Start with an empty project folder and create a file `app.jl`, which will be the entry point to the app, with the following content:
+
+```julia
+module App
+using GenieFramework
+@genietools
+
+end
+```
+The `GenieFramework` package includes everything you'll need to build the web app, and the `@genietools` macro enables features such as logging. To launch the app, start a Julia REPL with `julia --project` in the project folder and run
+
+```julia
+using GenieFramework; Genie.loadapp(); up()
+```
+
+This will start a server on port `8000` and you can access the app at `localhost:8000`. To stop the server, execute `down()` in the REPL.
+
+### Adding your Julia code
+
+Any local Julia modules or packages to be used in the app should be placed in the `lib` folder, whose content is automatically loaded by the server. In this example, we have a `StatisticAnalysis` module with functions to generate `N` random numbers and calulate the mean. Create the `lib` folder, and in it place a `StatisticAnalysis.jl` file with
+
+```julia [lib/StatisticAnalysis.jl]
+module StatisticAnalysis
+
+export gen_numbers, calc_mean
+
+function gen_numbers(N::Int)
+    return rand(N)
+end
+
+function calc_mean(x::Vector{Float64})
+    return round(sum(x) / length(x); digits=4)
+end
+end
+```
+
+Then, import the module in the main file as
+
+```julia [app.jl] {3,4}
+module App
+using GenieFramework
+include("lib/StatisticAnalysis.jl")
```

Comment:
OK, but you say that the code in `lib/` is automatically loaded, which is true... so why do you load it manually via `include`?